### PR TITLE
Setup POST API to Enqueue User

### PR DESCRIPTION
**Pull request by shishirbychapur**

### CODE REVIEW

File: backend/matching-service/src/types/UserQueueRequestDto.ts [hunk 1]
```diff
@@ -15,16 +19,17 @@ export class UserQueueRequestDto {
     @IsNotEmpty()
     userId: string
 
-    constructor(proficiency: Proficiency, topic: Category, userId: string) {
+    constructor(proficiency: Proficiency, difficulty: Complexity, topic: Category, userId: string) {
```
Comment (lynnlow175) [inline backend/matching-service/src/types/UserQueueRequestDto.ts:22]: maybe would be better to keep the naming same? like complexity: Complexity etc. so less confusion

Comment (Daviancold) [reply backend/matching-service/src/types/UserQueueRequestDto.ts:22]: i think using difficulty would help frontend understand what input to pass in for the api call, using the complexity enum

Comment (shishirbychapur) [reply backend/matching-service/src/types/UserQueueRequestDto.ts]: i think lets just standardize it as complexity (lol didnt see the second comment and i alr renamed everything haha)

**Review by lynnlow175 — APPROVED** [review]
lgtm

**Pull request merged**